test/com/linkedin/drelephant/mapreduce/heuristics/DistributedCacheLimitHeuristicTest.java

```diff
@@ -0,0 +1,137 @@
+package com.linkedin.drelephant.mapreduce.heuristics;
+
+import com.linkedin.drelephant.analysis.ApplicationType;
+import com.linkedin.drelephant.analysis.Heuristic;
+import com.linkedin.drelephant.analysis.HeuristicResult;
+import com.linkedin.drelephant.analysis.Severity;
+import com.linkedin.drelephant.configurations.heuristic.HeuristicConfigurationData;
+import com.linkedin.drelephant.mapreduce.data.MapReduceApplicationData;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+
+/**
+ * Tests for the <code>DistributedCacheLimitHeuristic</code> class.
+ */
+public class DistributedCacheLimitHeuristicTest {
+  private static Map<String, String> paramMap = new HashMap<String, String>();
+  private static Properties jobConf = new Properties();
+  private static final String cacheFileList =
+      "/path/to/firstCacheFile,/path/to/secondCacheFile,/path/to/thirdCacheFile";
+  private static final String archiveCacheFileList =
+      "/path/to/firstArchiveCacheFile,/path/to/secondArchiveCacheFile,/path/to/thirdArchiveCacheFile";
+
+  private static Heuristic<MapReduceApplicationData> _heuristic = new DistributedCacheLimitHeuristic(
+      new HeuristicConfigurationData("test.heuristic", "test.class", "test.view", new ApplicationType("mapreduce"),
+          paramMap));
+
+  @Before
+  public void setup() {
+    paramMap.put("distributed.cache.file.size.limit", "500000000");
+    jobConf.setProperty("mapreduce.job.cache.files", cacheFileList);
+    jobConf.setProperty("mapreduce.job.cache.archives", archiveCacheFileList);
+  }
+
+  /**
+   * All cache file sizes are within the limit.
+   */
+  @Test
+  public void testHeuristicResult() {
+    jobConf.setProperty("mapreduce.job.cache.files.filesizes", "100,200,300");
+    jobConf.setProperty("mapreduce.job.cache.archives.filesizes", "400,500,600");
```

Comment:
Are these tests passing ? you are specifying 500 here but I dont see it get multiplied with FILEUTILS*100MB befor checking against the limit